src/main/java/org/testng/internal/ClassHelper.java

```diff
@@ -224,20 +235,28 @@ public static IJUnitTestRunner createTestRunner(TestRunner runner) {
       isSamePackage = childPackage.getName().equals(classPackage.getName());
     }
 
-    for (Method method : declaredMethods) {
+    for (final Method method : declaredMethods) {
       int methodModifiers = method.getModifiers();
       if ((Modifier.isPublic(methodModifiers) || Modifier.isProtected(methodModifiers))
         || (isSamePackage && !Modifier.isPrivate(methodModifiers))) {
         if (!isOverridden(method, collected) && !Modifier.isAbstract(methodModifiers)) {
-          methods.add(method);
+          if (methods.containsKey(method.getName())) {
+            methods.get(method.getName()).add(method);
+            continue;
+          }
+          methods.put(method.getName(), new HashSet<Method>() {{add(method);}});
```

Comment:
@juherr - I didn't quite understand that part. That line will be executed only once when the map initially does not contain any entries for a particular method name. So its only once per method name. 
> Avoid {{}} because it is creating many anonymous sub-HashSet instances.

Can you please help me understand this comment ?